doc/src/arch/reference.rst

```diff
@@ -1897,6 +1902,98 @@ The clocking configuration is specified with ``<clock>`` tags within the ``<cloc
+        <clock_network  name="spine1" num_inst="2">
+            <spine metal_layer="global_spine"12 x="W/2" starty="0" endy="H">
+                <switch_point type="drive" name="drive" yoffset="H/2" buffer="drive_buff"/>
+                <switch_point type="tap" name="tap" yoffset="0" yincr="1"/>
+            </spine>
+        </clock_network>
+        <!-- Full Device: Each Grid Row -->
+        <clock_network  name="rib1" num_inst="2">
+            <rib metal_layer="global_rib" y="0" startx="0" endx="W" repeatx="W" repeaty="1">
+                <switch_point type="drive" name="drive" xoffset="W/2" buffer="drive_buff"/>
+                <switch_point type="tap" name="tap" xoffset="0" xincr="1"/>
+            </rib>
+        </clock_network>
+
+        <clock_routing>
+            <!-- connections from inter-block routing to central spine -->
+            <tap from="ROUTING" to="spine1.drive" locationx="W/2" locationy="H/2" switch="general_routing_switch" fc_val="1.0"/>
+            <!-- connections from spine to rib -->
+            <tap from="spine1.tap" to="rib1.drive" switch="general_routing_switch" fc_val="0.5"/>
+            <!-- connections from rib to clock pins -->
+            <tap from="rib1.tap" to="CLOCK" switch="ipin_cblock" fc_val="1.0"/>
+        </clock_routing >
+    </clocknetworks >
+
+The ``<metal_layers>`` element describes the per unit length electrical parameters used to implement the clock distribution  wires. There can be one or more wiring implementation options (metal layer, width and spacing) that are used by the later  clock network specification and each is described in a separate ``<metal_layer>`` sub-element. The syntax of the wiring electrical information is:
+
+.. arch:tag:: <metal_layer name="string" Rmetal="float" Cmetal="float"/>
+
+    :req_param name: A unique string for reference.
+    :req_param Rmetal: The resistance in Ohms of the wire per unit block in the FPGA architecture; a unit block usually corresponds to a logic cluster.
+    :req_pram Cmetal: The capacitance in Farads of the wire per unit block.
+
+The ``<clock_network>`` element contains sub-elements that describe the clock distribution wires for the clock architecture.  There could be more than one ``<clock_network>`` element to describe separate types of distribution wires. The high-level start tag for a clock network is as follows:
+
+.. arch:tag:: <clock_network name="string" num_inst="integer">
+
+    :req_param name: A unique string for reference.
+    :req_param num_inst: which describes the number of parallel instances of the clock distribution types described in the ``<clock_network>`` sub-elements.
+
+The supported clock distribution types are ``<spine>`` and ``<rib>``. ``Spine`` is used to describe vertical clock distribution wires. Whereas, ``Rib`` is used to describe a horizontal clock distribution wire.
+
+.. arch:tag:: <spine metal_layer="string" x="integer" starty="integer" endy="integer" repeatx="integer" repeaty="integer"/>
```

Comment:
Many of these parameters aren't actually integers but expressions (e.g. `H`, `H/2`, `W`). Should probably change the type to something like 'expr'.

We should also have a section which just describes what the expressions are and what the variables mean (e.g. W, H). If I recall correctly these are the same variables/expressions as used in grid construction, so we can likely just cross-ref to https://docs.verilogtorouting.org/en/latest/arch/reference/#grid-location-expressions. Although we should remind readers here that W and H are the device grid width/height and that `w` and `h` aren't available.